policies/tickets_expanded.rego

```diff
@@ -0,0 +1,95 @@
+package tickets_expanded
+
+import rego.v1
+
+roles := data.roles
+
+response.allow := allow
+
+response.reason := reason_admin
+
+response.conditions := conditions
+
+# NOTE(sr): Support both server/node and the other backends -- node sends the tenant information
+# differently.
+tenant := object.get(input, ["tenant", "name"], input.tenant)
+
+default allow := false
+
+allow if allowed(input.user, tenant, input.action)
+
+allowed(user, tenant, _) if "admin" in roles[tenant][user]
+
+allowed(user, tenant, action) if {
+	action in {"get", "list"}
+	some role in roles[tenant][user]
+	read_allowed(role)
+}
+
+allowed(user, tenant, "resolve") if user_is_resolver(user, tenant)
+
+read_allowed("reader")
+
+read_allowed("resolver")
+
+user_is_resolver(user, tenant) if "resolver" in roles[tenant][user]
+
+## CONDITIONS ##
+
+conditions["type"] := "compound"
+conditions["operation"] := "and"
+
+# Resolver conditions
+conditions["value"] contains {"type": "compound", "operation": "or", "value": v} if {
+	user_is_resolver(input.user, tenant)
+
+	v := [{"type": "compound", "operation": "and", "value": [
+			{"type": "field", "operation": "eq", "field": "tickets.resolved", "value": false},
+			{"type": "field", "operation": "eq", "field": "tickets.assignee", "value": null},
+		]},
+		{"type": "field", "operation": "eq", "field": "users.name", "value": input.user},
+	]
+}
```

Comment:
Would you think it would make sense to throw in a `ucast.expand()` at some later point, and make the conditions concise? 💭

Reply:
`tickets_expanded.rego` exists entirely because I wanted the same conditions as `tickets.rego`, just in the expanded form.

Currently, `ucast.expand` only exists for Enterprise OPA at this time, so we would need to add some kind of EOPA-specific switch, so that we don't accidentally run that policy under normal OPA, and error.

Reply:
:+1: Yeah right, I've forgotten that tickethub doesn't require EOPA.